volatility3/framework/plugins/windows/svcscan.py

```diff
@@ -150,13 +198,102 @@ def create_service_table(
             native_types=native_types,
         )
 
+    def _get_service_key(self, kernel):
+        for hive in hivelist.HiveList.list_hives(
+            context=self.context,
+            base_config_path=self.config_path,
+            layer_name=kernel.layer_name,
+            symbol_table=kernel.symbol_table_name,
+            filter_string="machine\\system",
+            hive_offsets=None,
+        ):
+            # Get ControlSet\Services.
+            try:
+                return hive.get_key(r"CurrentControlSet\Services")
+            except (KeyError, exceptions.InvalidAddressException):
+                try:
+                    return hive.get_key(r"ControlSet001\Services")
+                except (KeyError, exceptions.InvalidAddressException):
+                    pass
+
+            return None
+
+    @staticmethod
+    def _get_service_dll(
+        service_key,
+    ) -> Union[str, interfaces.renderers.BaseAbsentValue]:
+        try:
+            param_key = next(
```

Comment:
Quite an unusual syntax for doing what you're doing but... it works.  No real issue, just interesting to see it in use.  5:)